src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -57,6 +57,14 @@ Copyright (c) .NET Foundation. All rights reserved.
       PrepareForRazorCompile;
       RazorCoreCompile
     </RazorCompileDependsOn>
+
+  </PropertyGroup>
+
+  <!--
+    PreserveCompilationContext needs to enabled by default only for "applications" (OutputType = exe) which have one or more Razor files.
+  -->
+  <PropertyGroup Condition="'$(PreserveCompilationContext)' == '' AND '$(OutputType)' == 'exe'">
```

Comment:
You can't make property declaration condition on ItemGroup references in global scope. I filed https://github.com/aspnet/Razor/issues/2077 to track figuring out how to sort this out.

Reply:
Yeah that's right. bleh. OK